api/v1beta1/grafanamutetiming_types.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GrafanaMuteTimingSpec defines the desired state of GrafanaMuteTiming
+// +kubebuilder:validation:XValidation:rule="((!has(oldSelf.editable) && !has(self.editable)) || (has(oldSelf.editable) && has(self.editable)))", message="spec.editable is immutable"
+type GrafanaMuteTimingSpec struct {
+	GrafanaCommonSpec `json:",inline"`
+
+	// A unique name for the mute timing
+	Name string `json:"name"`
+
+	// Time intervals for muting
+	// +kubebuilder:validation:MinItems=1
+	TimeIntervals []*TimeInterval `json:"time_intervals"`
+
+	// Whether to enable or disable editing of the mute timing in Grafana UI
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="spec.editable is immutable"
+	// +optional
+	Editable *bool `json:"editable,omitempty"`
+}
+
+type TimeInterval struct {
+	// days of month
+	// +optional
+	DaysOfMonth []string `json:"days_of_month,omitempty"`
+
+	// location
+	// +optional
+	Location string `json:"location,omitempty"`
+
+	// months
+	// +optional
+	Months []string `json:"months,omitempty"`
+
+	// times
+	// +optional
+	Times []*TimeRange `json:"times,omitempty"`
+
+	// weekdays
+	// +optional
+	Weekdays []string `json:"weekdays,omitempty"`
+
+	// years
+	// +optional
+	Years []string `json:"years,omitempty"`
+}
```

Comment:
Validation is probably too difficult but the valid options from https://grafana.com/docs/grafana/latest/alerting/configure-notifications/mute-timings/#time-intervals would be nice as documentation strings